File: leetcode/11.container-with-most-water.py
# @leet start
class Solution:
    def maxArea(self, height: list[int]) -> int:
        best = 0
        left = 0
        right = len(height) - 1

        while left < right:
            best = max(best, min(height[left], height[right]) * (right - left))

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return best


# @leet end

File: leetcode/11.container-with-most-water.js
// @leet start
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let left = 0;
  let right = height.length - 1;
  let best = 0;

  while (left < right) {
    best = Math.max(
      best,
      (right - left) * Math.min(height[left], height[right]),
    );

    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return best;
};
// @leet end
